File: typescript/src/Wall/Brick.ts
export class Brick {
    message: string;
    creationDate: Date;

    constructor(message: string, creationDate: Date) {
        this.message = message;
        this.creationDate = creationDate;
    }

    getMessage(): string {
        return this.message;
    }

    getCreationDate(): Date {
        return this.creationDate;
    }
}

File: dotNet/src/Wall/Brick.cs
namespace src.Wall;

public class Brick
{
    public string Message { get; }
    public DateTime CreationDate { get; }

    public Brick(string message, DateTime creationDate)
    {
        Message = message;
        CreationDate = creationDate;
    }

    public override bool Equals(object obj)
    {
        if (this == obj)
            return true;

        if (!(obj is Brick brick))
            return false;

        return Message.Equals(brick.Message) && CreationDate.Equals(brick.CreationDate);
    }

    public override int GetHashCode()
    {
        int result = Message.GetHashCode();
        result = 31 * result + CreationDate.GetHashCode();
        return result;
    }

    public override string ToString()
    {
        return $"{nameof(Brick)}[message='{Message}', creationDate={CreationDate}]";
    }
}
